packages/hub/routes/card-spaces.ts

```diff
@@ -86,6 +86,20 @@ export default class CardSpacesRoute {
     ctx.type = 'application/vnd.api+json';
   }
 
+  async getUrlValidation(ctx: Koa.Context) {
+    if (!ensureLoggedIn(ctx)) {
+      return;
+    }
+
+    let url: string = ctx.params.url;
+    let errors = await this.cardSpaceValidator.validate({ url } as CardSpace);
+    let urlAvailable = !errors.url.includes('Already exists');
```

Comment:
Hm that's a good dilemma! 🤔 I wanted this field to be specifically tied to the fact whether this url has been taken or not. 

Now that I'm thinking about this, I think this is a bit of a roundabout solution...

it would be good to standardize this and let the client figure out what went wrong (from the error messages) and how to display them.

I'm thinking it would be better to simply respond with JSON:API compatible [error schema](https://jsonapi.org/examples/#error-objects-basics), like we did [here](https://github.com/cardstack/cardstack/blob/main/packages/hub/routes/card-spaces.ts#L63).

What do you think?

Reply:
I think that would be good. We can probably handle that in a separate task this sprint? There are other areas that should use this format, eg. the merchant slug/id validation.